### DISCUSSION TITLE
[SOLVED] github desktop cannot publish detached head lost changes

### DISCUSSION THREAD

**Discussion by cavigliera** [community/community#23339] [New to GitHub]
<p>Hallo,</p>
<p>yesterday i worked all the day on some php code.</p>
<p>Today i tried to put the new code in the repository and push on the web.</p>
<p>On Github desktop, i committed the changes but when i tried to push i got the error:</p>
<p>github desktop cannot publish detached head lost changes</p>
<p>I googled for this problem but wasn’t able to find something interesting.</p>
<p>Then i just clicked on current branch -&gt; master… and all my changes disappeared from the code folder!</p>
<p>No warnings or confirmation requests…</p>
<p>Is there a way to retreave my work?</p>
<p>Thanks a lot,</p>
<p>G.V.</p>
<p>[EDIT]</p>
<p>I found the solution myself:</p>
<p>in CLI tape </p>
<p>git reflog</p>
<p>In my case, the output was:</p>
<p>7d1c718 (HEAD -&gt; master, origin/master, origin/HEAD) HEAD@{0}: reset: moving to HEAD<br>
7d1c718 (HEAD -&gt; master, origin/master, origin/HEAD) HEAD@{1}: checkout: moving from f4c53154862813b4d536e29164a8bc97bc9a8c57 to master<br>
f4c5315 HEAD@{2}: commit: inizializzazione dubug phpstorm<br>
69bda05 HEAD@{3}: checkout: moving from master to 69bda057bfffc0b93ae58f0c4cb4e3d7809ba5d4<br>
69bda05 HEAD@{4}: reset: moving to 69bda057bfffc0b93ae58f0c4cb4e3d7809ba5d4<br>
2cf8a49 HEAD@{5}: revert: Revert “pre collegamento db pratiche”<br>
5d16dd3 HEAD@{6}: revert: Revert “Revert “pre collegamento db pratiche””<br>
51cb9f3 HEAD@{7}: revert: Revert “pre collegamento db pratiche”<br>
69bda05 HEAD@{8}: reset: moving to 69bda057bfffc0b93ae58f0c4cb4e3d7809ba5d4<br>
7d1c718 (HEAD -&gt; master, origin/master, origin/HEAD) HEAD@{9}: revert: Revert “pre collegamento db pratiche”<br>
b9ddc64 HEAD@{10}: revert: Revert “Revert “pre collegamento db pratiche””<br>
ce08bf1 HEAD@{11}: revert: Revert “pre collegamento db pratiche”<br>
69bda05 HEAD@{12}: commit: pre collegamento db pratiche<br>
46ec292 HEAD@{13}: commit: salvataggio prima modifica nome colonne relazioni<br>
a3bb260 HEAD@{14}: commit: inserimento fakedata doctrine<br>
3fd70fd HEAD@{15}: pull --progress origin: Fast-forward<br>
09cb040 HEAD@{16}: commit: ignored idea<br>
bc0d3db HEAD@{17}: commit: inizio migrazione classi in entita<br>
c113413 HEAD@{18}: pull --progress origin: Fast-forward<br>
dc65fee HEAD@{19}: commit: temp<br>
a1001f9 HEAD@{20}: clone: from <a href="https://github.com/cavigliera/fgcweb.git" rel="nofollow noopener">https://github.com/cavigliera/fgcweb.git</a></p>
<p>then i executed the command:</p>
<p>git branch recover-branch f4c53154862813b4d536e29164a8bc97bc9a8c57</p>
<p>where</p>
<ul>
<li>recover-branch is the name of the new branch we’ll create for recover</li>
</ul>
<p>- f4c53154862813b4d536e29164a8bc97bc9a8c57 is the sha on the output of “git reflog”.</p>
<p>Then i went on my ide, selected the new branch and recovered the last commit.</p>
<p>Hope this will help someone!</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/cavigliera">@cavigliera</a>, Thanks for coming back to share what worked for you. We’re looking forward to seeing you around!</p>
